### ISSUE TITLE
Read-only mode

### ISSUE THREAD

**Issue by edwardsnjd** [sallar/dropbox-fs#7]
*Thanks for this work, I tried and failed to get `BrowserFS` working over Dropbox then stumbled across `dropbox-fs` and had a prototype working in minutes!*

I have a toy project and am liable to make mistakes.  Naturally I should try to stay vigilant but having a read-only mode for `dropbox-fs` would give me some extra peace of mind.  What do you think?

I'm happy to have a go at a PR if you are interested.

---- 

Rather than polluting the existing implementation, I was thinking of adding a convenience wrapper that does something crude like replacing the dangerous methods with a no-op function that logs something.

Usage might look like:
```js
const dfs = require('dropbox-fs/readonly')({
    apiKey: 'DROPBOX_API_KEY_HERE',
    logger: console,
});
// e.g. dfs.writeFile(...) immediately succeeds without doing anything apart from logging (to the supplied `logger`)
```

Pseudo implementation:
```js
--- readonly.js ---
import normalDropboxFS from './index.js';

const dangerousMethods = [
   'writeFile',
   // etc.
];

exports default ({ logger=null, ...ctorOpts }) => {

    const api = normalDropboxFS(ctorOpts);

    const noop = (method) => (...methodArgs) => {
        if (logger) {
            // TODO: Some sort of logging of method and methodArgs
        }

        // Callback with success?
        const cb = methodArgs[methodArgs.length - 1];
        cb(null, null);  // Hmm, might be better to throw an exception?
    });

    // Replace dangerous methods with safe ones
    dangerousMethods.forEach(method => {
        api[method] = noop(method);
    });

    return api;
});
```

**Comment by sallar** [comment]
Hi @edwardsnjd, sorry for the late response.
This sounds very interesting. If you have time to do it and submit a PR, I'd be happy to have it merged :)

**Comment by edwardsnjd** [comment]
@sallar OK, great!  I've just pushed a PR (#8) now.

**Comment by sallar** [comment]
Closed by #8
